app/javascript/components/Study.js

```diff
@@ -1,46 +1,90 @@
+/* eslint-disable require-jsdoc */
 import React from 'react'
 
-const descriptionWordLimit = 100
+const descriptionWordLimit = 750
+const summaryWordLimit = 150
+const lengthOfHighlightTag = 21
 
 /* converts description into text snippet */
-export function formatDescription(rawDescription) {
+export function formatDescription(rawDescription, term) {
   const textDescription = stripTags(rawDescription)
-  return shortenDescription(textDescription)
+  return shortenDescription(textDescription, term)
 }
 
-// return the first 100 words, with a '...' appended if needed'
-// this returns a node, not text, so we can italicize 'continued'
-function shortenDescription(textDescription) {
-  const wordArray = textDescription.split(' ')
-  const numWords = wordArray.length
-  // take the first 100 words, or the whole thing if shorter
-  const shortenedText = wordArray.slice(0, descriptionWordLimit - 1)
-                                 .join(' ')
-
-  let suffixTag = <></>
-  if (numWords > descriptionWordLimit) {
-    suffixTag = <span className="detail"> ...(continued)</span>
+function highlightText(text, terms) {
+  const matchedIndicies = []
+  if (terms) {
+    let match
+    const regex = RegExp(terms, 'gi')
```

Comment:
I don't think this will do what you expect when there are multiple terms.  Passing an array ['a','b'] to RegExp will only return matches on the string 'a,b'